modules/tensor_mechanics/include/materials/DiscreteRRIsotropicPlasticity.h

```diff
@@ -0,0 +1,62 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef DISCRETERRISOTROPICPLASTICITY_H
+#define DISCRETERRISOTROPICPLASTICITY_H
+
+#include "DiscreteRadialReturnStressIncrement.h"
+#include "PiecewiseLinear.h"
+#include "MooseMesh.h"
+
+// forward declarations
+class PiecewiseLinear;
+
+/**
+ */
+
+class DiscreteRRIsotropicPlasticity : public DiscreteRadialReturnStressIncrement
+{
+public:
+  DiscreteRRIsotropicPlasticity( const InputParameters & parameters);
+
+protected:
+  void resetQpProperties(); // Values here need to be set to a constant, ideally zero
+
+  virtual void computeStressInitialize(Real effectiveTrialStress);
+  virtual Real computeResidual(Real effectiveTrialStress, Real scalar);
+  virtual Real computeDerivative(Real effectiveTrialStress, Real scalar);
+  virtual void iterationFinalize(Real scalar);
+  virtual void computeStressFinalize(const RankTwoTensor & plasticStrainIncrement);
```

Comment:
@idaholab/moose-users  How strict should we be on doxygen comments? Generally, I personally like to see everything in the header have a doxygen comment, even if it is obvious. However, I am not sure that this is something that should be enforced universally, what are peoples feelings?